react/ProductSummaryContext.tsx

```diff
@@ -33,14 +35,23 @@ type SetLoadingAction = {
   }
 }
 
-type Action = SetProductAction | SetHoverAction | SetLoadingAction
+type SetProductQuantity = {
+  type: 'SET_QUANTITY'
+  args: {
+    quantity: number
+  }
+}
+
+type Action = SetProductAction | SetHoverAction | SetLoadingAction | SetProductQuantity
 
 export function reducer(state: State, action: Action) {
   switch (action.type) {
     case 'SET_PRODUCT': {
+      const product = action.args.product
       return {
         ...state,
-        product: action.args.product,
+        product: product,
+        selectedItem: product.sku
```

Comment:
Get selectedItem based on the first sku with available quantity.

Reply:
similar logic is used in ProductWrapper in `store` app, check there